### DISCUSSION TITLE
Runtime dependencies?

### DISCUSSION THREAD

**Discussion by diamondap** [ncruces/zenity#45] [Q&A] [answered]
The project readme lists dependencies for Mac and Unix/Linux. Are these build-time dependencies, or runtime dependencies?

To put it another way, if I use zenity and distribute my app, do I have to tell Mac & *nix users they must install osascript or zenity to run my app?

**Comment by ncruces** [comment, marked as answer]
They are runtime dependencies.

Unix/Linux users need to install `zenity` (or a clone, like [`qarma`](https://github.com/luebking/qarma)).
`osascript` (and its sister `osacompile`, which is needed for the progress dialog) is preinstalled on every mac.

You can test for the availability of runtime dependencies using [`zenity.IsAvailable()`](https://pkg.go.dev/github.com/ncruces/zenity#IsAvailable).

**Reply by diamondap** [reply]
Thanks for clarifying. So it sounds like Linux/Unix is the only platform where users might need to take an extra step to install a dependency. That should be fine for our use case.

Thanks again for your help!
